core/internal/filetransfer/file_transfer_default.go

```diff
@@ -102,10 +102,8 @@ func (ft *DefaultFileTransfer) Upload(task *Task) error {
 		}
 		req.Header.Set(parts[0], parts[1])
 	}
-	if _, err := ft.client.Do(req); err != nil {
-		return err
-	}
-	return nil
+	_, err = ft.client.Do(req)
+	return err
```

Comment:
why?

Reply:
Oh, just the extra `if` bothered me. I originally had changes here so it made sense, but I can revert this file since it's not a behavior change.

Reply:
i guess it is logically equivalent ... so up to you